src/chess/board.h

```diff
@@ -62,8 +62,6 @@ class ChessBoard {
   ChessBoard() = default;
   ChessBoard(const std::string& fen) { SetFromFen(fen); }
 
-  ChessBoard& operator=(const ChessBoard&) = default;
```

Comment:
Wasn't this required for https://github.com/LeelaChessZero/lc0/pull/1942?

Reply:
Apple's clang complained about it:
```
../src/chess/board.h:65:15: warning: definition of implicit copy constructor for 'ChessBoard' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
  ChessBoard& operator=(const ChessBoard&) = default;
  ```

Reply:
But how does it compile without this function? :thinking: 

Anyway, I think what they kind of suggest is to also add a default copy constructor:
```cpp
ChessBoard(const ChessBoard&) = default;
```

Reply:
This works as well.